### PR TITLE
Remove unused locals and methods in System.Private.Xml

### PR DESCRIPTION
**Pull request by MattKotsenas**
Remove unused locals and private methods in System.Private.Xml.

This fixes part of #30457

### CODE REVIEW

**Comment by krwq** [comment]
Please do not merge this PR until we're done with nullable annotations for XML. Merging may cause unnecessary conflicts with work in progress which we're currently targeting for 5.0, you can track the effort here: https://github.com/dotnet/runtime/issues/2339

**Comment by stephentoub** [comment]
@MattKotsenas, can you rebase this on master, now that the PRs @krwq described have been merged?  Thanks.

**Comment by MattKotsenas** [comment]
@stephentoub, rebased. @krwq, can you take another look when you have a moment? Thanks!

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs [hunk 1]
```diff
@@ -186,8 +186,8 @@ protected XmlDocument Document
         private void InitPrimitiveIDs()
         {
             if (_tokenID != null) return;
-            object ns = _r.NameTable.Add(XmlSchema.Namespace);
-            object ns2 = _r.NameTable.Add(UrtTypes.Namespace);
+            _ = _r.NameTable.Add(XmlSchema.Namespace);
+            _ = _r.NameTable.Add(UrtTypes.Namespace);
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs:190]:
```suggestion
            _r.NameTable.Add(XmlSchema.Namespace);
            _r.NameTable.Add(UrtTypes.Namespace);
```

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/CompilerScopeManager.cs [hunk 2]
```diff
@@ -274,8 +274,8 @@ public bool IsExNamespace(string nsUri)
                     bool undefined = false;
                     for (int prev = record + 1; prev < exAll; prev++)
                     {
-                        string? prevPrefix, prevNsUri;
-                        ScopeFlags prevFlags = GetName(ref _records[prev], out prevPrefix, out prevNsUri);
+                        string? prevPrefix;
+                        _ = GetName(ref _records[prev], out prevPrefix, out _);
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/CompilerScopeManager.cs:278]:
```suggestion
                        GetName(ref _records[prev], out prevPrefix, out _);
```

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs [hunk 3]
```diff
@@ -844,7 +844,7 @@ private void LoadKey(NsDecl? stylesheetNsList)
             QilName keyName = ParseQNameAttribute(0)!;
             string? match = ParseStringAttribute(1, "match");
             string? use = ParseStringAttribute(2, "use");
-            string? collation = ParseCollationAttribute(3);
+            _ = ParseCollationAttribute(3);
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs:847]:
```suggestion
            ParseCollationAttribute(3);
```

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs [hunk 4]
```diff
@@ -2138,7 +2137,7 @@ private XslNode XslSort(int sortNumber)
             string? select = ParseStringAttribute(0, "select");
             string? lang = ParseStringAttribute(1, "lang");
             string? order = ParseStringAttribute(2, "order");
-            string? collation = ParseCollationAttribute(3);
+            _ = ParseCollationAttribute(3);
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs:2140]:
```suggestion
            ParseCollationAttribute(3);
```

File: src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/CreateCDataSectionTests.cs [hunk 5]
```diff
@@ -37,7 +37,7 @@ public static void CreateCDataTest3()
             var xmlDocument = new XmlDocument();
             var cdata = xmlDocument.CreateCDataSection("]]>");
 
-            AssertExtensions.Throws<ArgumentException>(null, () => { var test = cdata.OuterXml; });
+            AssertExtensions.Throws<ArgumentException>(null, () => { _ = cdata.OuterXml; });
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/CreateCDataSectionTests.cs:40]:
```suggestion
            AssertExtensions.Throws<ArgumentException>(null, () => cdata.OuterXml);
```

File: src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs [hunk 6]
```diff
@@ -741,7 +741,7 @@ public static void Xml_DeserializeOutOfRangeByteProperty()
             writer.Flush();
             stream.Position = 0;
             Assert.Throws<InvalidOperationException>(() => {
-                var deserializedObj = (TypeWithByteProperty)serializer.Deserialize(stream);
+                _ = (TypeWithByteProperty)serializer.Deserialize(stream);
             });
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs:745]:
```C#
Assert.Throws<InvalidOperationException>(() => (TypeWithByteProperty)serializer.Deserialize(stream));
```

File: src/libraries/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderConstructionTests.cs [hunk 7]
```diff
@@ -22,7 +22,7 @@ public void NodeReaderConstructionWithNull()
         {
             Assert.Throws<ArgumentNullException>(() =>
             {
-                var nodeReader = new XmlNodeReader(null);
+                new XmlNodeReader(null);
             });
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderConstructionTests.cs:26]:
```C#
Assert.Throws<ArgumentNullException>(() => new XmlNodeReader(null));
```

File: src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/NameTableTests.cs [hunk 8]
```diff
@@ -31,7 +31,7 @@ public static void RespectManuallyAddedReference()
         public static void RespectTypesDerivedFromNameTable()
         {
             var customNameTable = new CustomNameTable();
-            var xmlDocument = new XmlDocument(customNameTable);
+            _ = new XmlDocument(customNameTable);
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/NameTableTests.cs:34]:
```suggestion
            new XmlDocument(customNameTable);
```

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs (hunk 1 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs]: Fixed!

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/CompilerScopeManager.cs (hunk 2 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/CompilerScopeManager.cs]: Fixed!

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs (hunk 3 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs]: Fixed!

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs (hunk 4 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs]: Fixed!

File: src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/CreateCDataSectionTests.cs (hunk 5 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/CreateCDataSectionTests.cs]: Fixed!

File: src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs (hunk 6 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs]: Fixed!

File: src/libraries/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderConstructionTests.cs (hunk 7 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderConstructionTests.cs]: Fixed!

File: src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/NameTableTests.cs (hunk 8 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/NameTableTests.cs]: Fixed!

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/ContentValidator.cs [hunk 9]
```diff
@@ -1536,50 +1534,6 @@ private void CheckUniqueParticleAttribution(BitSet curpos)
             // now convert transition table to array
             return (int[][])transitionTable.ToArray(typeof(int[]));
         }
-
-#if DEBUG
-        private void Dump(StringBuilder bb, BitSet[] followpos, int[][] transitionTable)
```
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Schema/ContentValidator.cs:1541]: I've occasionally found the DEBUG code useful when debugging some cryptic bugs in XML. I'd prefer we left it in place

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs [hunk 10]
```diff
@@ -2439,98 +2439,6 @@ private ContentValidator CompileComplexContent(XmlSchemaComplexType complexType)
             }
         }
 
-#if DEBUG
```
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs:2442]: I'd prefer we left this debug code in here as it is occasionally useful

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/ContentValidator.cs (hunk 9 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Schema/ContentValidator.cs]: Restored. Thanks for the feedback!

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs (hunk 10 above)
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs]: Restored! Thanks for the feedback!

**Comment by MattKotsenas** [comment]
I don't have merge permissions, so if everything looks good, please merge on my behalf. If there are any issues or questions, please let me know. Thanks!

**Review by krwq — APPROVED** [review]

**Comment by krwq** [comment]
Thanks @MattKotsenas, looks good to me, let's see if we can get clean CI run first

**Comment by danmosemsft** [comment]
@krwq  you can set auto-merge label in such a case of course

**Comment by krwq** [comment]
Thanks @MattKotsenas!

**Pull request merged**